Sources/NIOCore/BSDSocketAPI.swift

```diff
@@ -245,6 +245,12 @@ extension NIOBSDSocket.OptionLevel {
     /// Socket options that apply to all sockets.
     public static let socket: NIOBSDSocket.OptionLevel =
             NIOBSDSocket.OptionLevel(rawValue: SOL_SOCKET)
+
+    #if os(Linux)
+    /// Socket options that apply only to UDP sockets.
+    public static let sol_udp: NIOBSDSocket.OptionLevel =
```

Comment:
Per the naming scheme, I think this is just `udp`. Also, prefer `IPPROTO_UDP` to `SOL_UDP`. That way we can also make this available on Darwin and Windows.

Reply:
Will change this. What's the difference between `IPPROTO_UDP` and `SOL_UDP` all of the references and docs I found referred to the latter.

Reply:
In terms of its value, literally nothing: both evaluate to 17. `SOL_UDP` is kinda a bastardisation: the `SOL_TCP` and `SOL_UDP` are both aliases to `IPPROTO_TCP` and `IPPROTO_UDP`, but the latter two spellings are more common.